lib/formats/odata.js

```diff
@@ -464,7 +475,16 @@ const filterFields = (formFields, select, table) => {
       }
     }
   }
-  return Array.from(filteredFields);
+
+  const filteredFieldsArr = Array.from(filteredFields);
+  // Sort the fields by `order` as Set() doesn't gurantee an order
```

Comment:
I think `Set`s and `Map`s iterate according to insertion order, in case that helps!

Reply:
insertion order is based on the value of `$select`, which is defined by the user. So I have to order it here